File: CarlaErik/UvA.SoftCon.Questionnaire/AST/Model/Expressions/Literals/BooleanLiteral.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UvA.SoftCon.Questionnaire.AST.Model.Statements;

namespace UvA.SoftCon.Questionnaire.AST.Model.Expressions.Literals
{
    /// <summary>
    /// Represents a static, immutable boolean value.
    /// </summary>
    public class BooleanLiteral : Literal<bool>
    {
        public override NodeType Type
        {
            get
            {
                return NodeType.BooleanLiteral;
            }
        }

        public BooleanLiteral(bool value, TextPosition position)
            : base(value, position)
        {
        }

        public override T Accept<T>(IASTVisitor<T> visitor)
        {
            return visitor.Visit(this);
        }

        public override DataType? GetType(IDictionary<string, DataType> symbolTable)
        {
            return DataType.Boolean;
        }
    }
}

File: Nian_Karl/QL_Nian_Karl/src/org/uva/sea/ql/AST/literal/BooleanLiteral.java
package org.uva.sea.ql.AST.literal;

import org.uva.sea.ql.AST.visitor.Visitor;


public class BooleanLiteral extends AbstractLiteral{
	private final boolean booleanValue;

	public BooleanLiteral(boolean boolValue) {
		this.booleanValue = boolValue;
	}
	
	@Override
	public void accept(Visitor visitor) {
		visitor.visit(this);		
	}
	
}
